File: services/categoria-service.ts
import { collection, getDocs, doc, getDoc } from "firebase/firestore"
import { db } from "../firebase-config"

// Interfaz para categorías
export interface Categoria {
  id: string
  nombre: string
  nombreKichwa: string
  descripcion: string
  icono: string
}

// Obtener todas las categorías
export const getCategorias = async (): Promise<Categoria[]> => {
  try {
    const categoriasRef = collection(db, "categorias")
    const snapshot = await getDocs(categoriasRef)

    if (snapshot.empty) {
      console.log("No hay categorías disponibles")
      return categoriasEjemplo // Devolver datos de ejemplo si no hay datos reales
    }

    return snapshot.docs.map((doc) => ({
      id: doc.id,
      ...doc.data(),
    })) as Categoria[]
  } catch (error) {
    console.error("Error al obtener categorías:", error)
    return categoriasEjemplo // Devolver datos de ejemplo en caso de error
  }
}

// Obtener una categoría por ID
export const getCategoriaById = async (categoriaId: string): Promise<Categoria | null> => {
  try {
    const categoriaRef = doc(db, "categorias", categoriaId)
    const categoriaDoc = await getDoc(categoriaRef)

    if (!categoriaDoc.exists()) {
      console.log("No se encontró la categoría")
      return null
    }

    return {
      id: categoriaDoc.id,
      ...categoriaDoc.data(),
    } as Categoria
  } catch (error) {
    console.error("Error al obtener categoría:", error)
    return null
  }
}

// Datos de ejemplo para categorías (se usan cuando no hay datos en Firestore)
export const categoriasEjemplo: Categoria[] = [
  {
    id: "c1",
    nombre: "Limpieza",
    nombreKichwa: "Pichana",
    descripcion: "Servicios de limpieza para hogares y oficinas",
    icono: "/placeholder.svg?height=40&width=40",
  },
  {
    id: "c2",
    nombre: "Plomería",
    nombreKichwa: "Yaku Ruray",
    descripcion: "Reparación e instalación de sistemas de agua",
    icono: "/placeholder.svg?height=40&width=40",
  },
  {
    id: "c3",
    nombre: "Electricidad",
    nombreKichwa: "Achik Ruray",
    descripcion: "Instalación y reparación de sistemas eléctricos",
    icono: "/placeholder.svg?height=40&width=40",
  },
  {
    id: "c4",
    nombre: "Carpintería",
    nombreKichwa: "Kaspi Ruray",
    descripcion: "Trabajos en madera y muebles",
    icono: "/placeholder.svg?height=40&width=40",
  },
  {
    id: "c5",
    nombre: "Jardinería",
    nombreKichwa: "Pampa Kamay",
    descripcion: "Mantenimiento de jardines y áreas verdes",
    icono: "/placeholder.svg?height=40&width=40",
  },
  {
    id: "c6",
    nombre: "Cuidado",
    nombreKichwa: "Kamana",
    descripcion: "Cuidado de niños, adultos mayores y mascotas",
    icono: "/placeholder.svg?height=40&width=40",
  },
]

File: lib/services/categoria_service.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import '../models/categoria.dart';

class CategoriaService {
  final FirebaseFirestore _firestore = FirebaseFirestore.instance;

  // Obtener todas las categorías
  Future<List<Categoria>> getCategorias() async {
    try {
      final snapshot = await _firestore.collection('categorias').get();

      if (snapshot.docs.isEmpty) {
        print('No hay categorías disponibles');
        return categoriasEjemplo;
      }

      return snapshot.docs
          .map((doc) => Categoria.fromMap(doc.data(), doc.id))
          .toList();
    } catch (e) {
      print('Error al obtener categorías: $e');
      return categoriasEjemplo;
    }
  }

  // Obtener una categoría por ID
  Future<Categoria?> getCategoriaById(String categoriaId) async {
    try {
      final doc = await _firestore.collection('categorias').doc(categoriaId).get();

      if (!doc.exists) {
        print('No se encontró la categoría');
        return null;
      }

      return Categoria.fromMap(doc.data()!, doc.id);
    } catch (e) {
      print('Error al obtener categoría: $e');
      return null;
    }
  }
}

// Datos de ejemplo para categorías
final List<Categoria> categoriasEjemplo = [
  Categoria(
    id: 'c1',
    nombre: 'Limpieza',
    nombreKichwa: 'Pichana',
    descripcion: 'Servicios de limpieza para hogares y oficinas',
    icono: 'assets/images/categoria_limpieza.png',
  ),
  Categoria(
    id: 'c2',
    nombre: 'Plomería',
    nombreKichwa: 'Yaku Ruray',
    descripcion: 'Reparación e instalación de sistemas de agua',
    icono: 'assets/images/categoria_plomeria.png',
  ),
  Categoria(
    id: 'c3',
    nombre: 'Electricidad',
    nombreKichwa: 'Achik Ruray',
    descripcion: 'Instalación y reparación de sistemas eléctricos',
    icono: 'assets/images/categoria_electricidad.png',
  ),
  Categoria(
    id: 'c4',
    nombre: 'Carpintería',
    nombreKichwa: 'Kaspi Ruray',
    descripcion: 'Trabajos en madera y muebles',
    icono: 'assets/images/categoria_carpinteria.png',
  ),
  Categoria(
    id: 'c5',
    nombre: 'Jardinería',
    nombreKichwa: 'Pampa Kamay',
    descripcion: 'Mantenimiento de jardines y áreas verdes',
    icono: 'assets/images/categoria_jardineria.png',
  ),
  Categoria(
    id: 'c6',
    nombre: 'Cuidado',
    nombreKichwa: 'Kamana',
    descripcion: 'Cuidado de niños, adultos mayores y mascotas',
    icono: 'assets/images/categoria_cuidado.png',
  ),
];
